test/unit/rubygem_test.rb

```diff
@@ -683,5 +683,32 @@ class RubygemTest < ActiveSupport::TestCase
         assert_equal(("01".."30").map { |date| "10/#{date}" }, Rubygem.monthly_short_dates)
       end
     end
+
+    context "give most downloaded gems" do
+      setup do
+        @thin = create(:rubygem, :name => 'thin', :downloads => 76)
+        @rake = create(:rubygem, :name => 'rake', :downloads => 45)
+        @json = create(:rubygem, :name => 'json', :downloads => 32)
+        @thor = create(:rubygem, :name => 'thor', :downloads => 20)
+        @rack = create(:rubygem, :name => 'rack', :downloads => 13)
+        @dust = create(:rubygem, :name => 'dust', :downloads => 0)
+        @haml = create(:rubygem, :name => 'haml', :downloads => 0)
+        @chronic = create(:rubygem, :name => 'chronic', :downloads => 0)
+
+        @gems = [@thin, @rake, @json, @thor, @rack, @dust, @haml, @chronic]
+        @gems.each do |g|
+          create(:version, :rubygem => g)
+          g[:downloads].times { Download.incr(g.name, @version.full_name) }
+        end
+      end
+
+      should "by postgres" do
+        assert_equal [@thin, @rake, @json, @thor, @rack], Rubygem.most_downloaded_by_db(5)
+      end
+
+      should "by redis" do
+        assert_equal [@thin, @rake, @json, @thor, @rack], Rubygem.most_downloaded_by_redis(5)
+      end
+    end
```

Comment:
I would trust these tests more if the ranking was different for Redis vs. Postgres.